builtin/providers/google/resource_compute_backend_service.go

```diff
@@ -0,0 +1,410 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func resourceComputeBackendService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeBackendServiceCreate,
+		Read:   resourceComputeBackendServiceRead,
+		Update: resourceComputeBackendServiceUpdate,
+		Delete: resourceComputeBackendServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": &schema.Schema{
+				Type: schema.TypeSet,
```

Comment:
I just thought it makes more sense for all singleton fields, but it has apparently more issues than advantages.

Except the extra effort with hash function you mentioned, it's very difficult to reference these fields, since the attribute key would be something like `backend.1178273152.balancing_mode`, whereas with `TypeList` it's `backend.0.balancing_mode` which is much more predictable.

It would be nice if it was `backend.balancing_mode`, but that's another story...